src/crewai/agent.py

```diff
@@ -114,39 +114,42 @@ class Agent(BaseAgent):
     )
 
     @model_validator(mode="after")
-    def set_agent_ops_agent_name(self) -> "Agent":
-        """Set agent ops agent name."""
+    def post_init_setup(self):
+        # Set agent_ops_agent_name
         self.agent_ops_agent_name = self.role
-        return self
 
-    @model_validator(mode="after")
-    def set_agent_executor(self) -> "Agent":
-        """Ensure agent executor and token process are set."""
+        # Set up LLM callbacks
         if hasattr(self.llm, "model_name"):
```

Comment:
We do a bunch of things in `_setup_llm_callbacks` that do not require a `model_name` to be present in `model`. Should we move this check in?

Reply:
Hey! Good question.

Under the hood, we need the model name because tiktoken uses that model name to calculate tokens. 

If we don't have the model name, we can't calculate tokens.

Reply:
Right, but what about that thing that makes sure `callbacks` is a list, just to name one?